### ISSUE TITLE
[Bug]: @actual-app/api is not syncing imported transactions to server

### ISSUE THREAD

**Issue by bowdi** [actualbudget/actual#1226]
### Verified issue does not already exist?

- [X] I have searched and found no existing issue

### What happened?

I've got a node / TS project that imports transactions from my bank into actual but it doesn't seem to actually save the transactions.

I'm using `.importTransactions()` to import the transactions. It does not throw any error messages and it even returns id's for transactions it has added.

The transactions just never make it to the server.

I've verified both direction of network traffic and it has not problem communicating to the server or downloading the files.

I even pulled the db.sqlite from the project server and ran the below

```
SELECT id, account, amount, notes, date, imported_id, imported_payee FROM v_transactions ORDER BY date DESC LIMIT 10;
```

and still could not find the transactions there which is even more weird as it's like it didn't save them locally either.

The server and this project are running on docker (compose).

I tried to add some debug flags `DEBUG=actual:*` to both the server and the project but nothing useful comes up

Other info:
project docker container base image: `node:16-alpine`
project api client version: `6.1.0`
server image: `actualbudget/actual-server:23.6.0`

Let me know if you need any other info / logs

### What error did you receive?

None

### Where are you hosting Actual?

Docker

### What browsers are you seeing the problem on?

Other

### Operating System

Linux

**Comment by j-f1** [comment]
If you call `api.internal.send('sync')` before shutting it down does it work?

**Comment by phootip** [comment]
> If you call `api.internal.send('sync')` before shutting it down does it work?

It does work, and I believe in the next api version we could just use `api.sync()` instead

**Comment by bowdi** [comment]
So thank you for those as I have been looking forward to the release of that api.sync method!

I actually managed to find out it was because I was downloading two budgets and thought the api would be able to assign transactions to the correct one, it just failed silently though so perhaps some validation is needed?

I debugged all the way down to the `apply` methods and could see the transactions were generating insert + update queries without error's which is weird!
